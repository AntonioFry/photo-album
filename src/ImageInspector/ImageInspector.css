.faded-background {
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  position: absolute;
  top: 0px;
  height: 100%;
  width: 100%;
}

.image-container {
  transition: padding 0.25s linear;
  height: 400px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0px 0px 0px 0px;
}

.image-container:hover .image-inspector-comment-container {
  visibility: visible;
}

.inspected-image {
  width: 600px;
}

.image-inspector-comment-container {
  background-color: rgba(247, 247, 247, 0.5);
  bottom: 134px;
  padding: 10px;
  position: absolute;
  width: 600px;
  visibility: hidden;
}

.image-inspector-photo-comment {
  margin: 0px;
}

.right-button,
.left-button {
  transition: color 0s linear,
              margin 0.25s linear,
              width 0.25s linear;
  background-color: rgba(0, 0, 0, .65);
  border: none;
  color: transparent;
  padding: 0px;
  height: 80px;
  width: 10px;
}

.right-button:hover,
.left-button:hover {
  transition: margin 0.25s linear,
              color 0.25s step-end,
              width 0.25s linear;
  background-color: rgba(0, 0, 0, .8);
  color: white;
  width: 30px;
}

.right-button:hover {
  margin: 0px
}

.left-button:hover {
  margin: 0px;
}

.right-button {
  border-top-right-radius: 7px;
  border-bottom-right-radius: 7px;
  margin: 0px 20px 0px 0px;
}

.left-button {
  border-top-left-radius: 7px;
  border-bottom-left-radius: 7px;
  margin: 0px 0px 0px 20px;
}

.right-button:disabled,
.left-button:disabled {
  width: 10px;
  color: transparent;
  background-color: rgba(0, 0, 0, 0.3);
}

.right-button:disabled {
  margin: 0px 20px 0px 0px;
}

.left-button:disabled {
  margin: 0px 0px 0px 20px;
}

.close-button {
  background-color: rgba(0, 0, 0, 0.65);
  border: none;
  color: white;
  position: relative;
  bottom: 185px;
  left: 630px;
  text-align: center;
  width: 30px;
  height: 30px;
  z-index: 2;
}

.close-button:hover {
  background-color: rgba(0, 0, 0, 0.8);
}