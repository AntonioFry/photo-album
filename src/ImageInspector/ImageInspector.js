import React, { Component } from 'react';
import './ImageInspector.css';

class ImageInspector extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageCollection: props.imageCollection || [],
      currentImageIndex: 0,
      currentImage: props.startingPhoto,
    }
  }

  changePhoto = async (direction) => {
    const { imageCollection, currentImageIndex } = this.state;
    if (direction === 'left') {
      await this.setState({ currentImageIndex: currentImageIndex - 1 });
      this.setState({ currentImage: imageCollection[this.state.currentImageIndex].imageSrc });
    } else if (direction  === 'right') {
      await this.setState({ currentImageIndex: currentImageIndex + 1 });
      this.setState({ currentImage: imageCollection[this.state.currentImageIndex].imageSrc });
    }
  }

  render() {
    const { currentImage, imageCollection, currentImageIndex } = this.state;

    return (
      <section className="faded-background">
        <button data-testid="close-btn" className="close-button" onClick={() => this.props.closeInspector()}>X</button>
        <button
          data-testid="left-btn"
          className='left-button' 
          onClick={() => this.changePhoto('left')} 
          disabled={imageCollection[currentImageIndex - 1] === undefined}>
            {'<'}
        </button>
        <div className="image-container">
          <img data-testid="inspected-img" className='inspected-image' src={currentImage} alt={imageCollection[currentImageIndex].photoDescription}/>
          <article className="image-inspector-comment-container">
            <p data-testid="photo-comment" className="image-inspector-photo-comment">{imageCollection[currentImageIndex].photoComment}</p>
          </article>
        </div>
        <button
          data-testid="right-btn"
          className='right-button'
          onClick={() => this.changePhoto('right')}
          disabled={imageCollection[currentImageIndex + 1] === undefined}>
            {'>'}
        </button>
      </section>
    )
  }
};

export default ImageInspector;