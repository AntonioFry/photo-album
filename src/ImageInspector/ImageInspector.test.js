import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import ImageInspector from './ImageInspector';
import renderer from 'react-test-renderer';

describe('ImageInspector', () => {
  let mockData;

  beforeEach(() => {
    mockData = {
      albumName: 'Bird Photos',
      photos: [
        {
          imageSrc: require('../Data/Images/grey-jay.JPG'),
          photoComment: "Took this shot up at Echo Lake when the bird landed meer feet away from me. It wasn't until I searched up th bird I realized it was a grey jay.",
          photoDescription: "Grey Jay perched and looking out"
        },
        {
          imageSrc: require('../Data/Images/blue-scrub-jay.JPG'),
          photoComment: "Though they are common in Southern Colorado, they really stand out to me because of their color",
          photoDescription: "Blue scrub jay perched and looking out"
        }
      ]
    }
  });

  it('Should render correctly', () => {
    const { getByTestId } = render(<ImageInspector 
      imageCollection={mockData.photos}
      startingPhoto={mockData.photos[0]}
      closeInspector={() => jest.fn()}
    />)
    expect(getByTestId('photo-comment')).toHaveTextContent(mockData.photos[0].photoComment)
  });

  it('Should match Snapshot', () => {
    const tree = renderer.create(<ImageInspector
      imageCollection={mockData.photos}
      startingPhoto={mockData.photos[0]}
      closeInspector={() => jest.fn()}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  })

})