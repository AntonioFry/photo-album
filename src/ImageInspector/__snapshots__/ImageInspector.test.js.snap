// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageInspector Should match Snapshot 1`] = `
<section
  className="faded-background"
>
  <button
    className="close-button"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="left-button"
    disabled={true}
    onClick={[Function]}
  >
    &lt;
  </button>
  <div
    className="image-container"
  >
    <img
      alt="Grey Jay perched and looking out"
      className="inspected-image"
      src={
        Object {
          "imageSrc": "grey-jay.JPG",
          "photoComment": "Took this shot up at Echo Lake when the bird landed meer feet away from me. It wasn't until I searched up th bird I realized it was a grey jay.",
          "photoDescription": "Grey Jay perched and looking out",
        }
      }
    />
    <article
      className="image-inspector-comment-container"
    >
      <p
        className="image-inspector-photo-comment"
        data-testid="photo-comment"
      >
        Took this shot up at Echo Lake when the bird landed meer feet away from me. It wasn't until I searched up th bird I realized it was a grey jay.
      </p>
    </article>
  </div>
  <button
    className="right-button"
    disabled={false}
    onClick={[Function]}
  >
    &gt;
  </button>
</section>
`;
